tests/apidoc_profile_displayname_test.go

```diff
@@ -0,0 +1,46 @@
+package tests
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/internal/match"
+	"github.com/matrix-org/complement/internal/must"
+)
+
+func TestProfileDisplayName(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	unauthedClient := deployment.Client(t, "hs1", "")
+	authedClient := deployment.RegisterUser(t, "hs1", "display_name_user", "superuser")
```

Comment:
Why? We aren't testing registration/login anymore where we need to do this. You have a built-in "alice" client available, use it.